source/_integrations/recorder.markdown

```diff
@@ -362,6 +362,12 @@ pip3 install mysqlclient
 
 After installing the dependencies, it is required to create the database manually. During the startup, Home Assistant will look for the database specified in the `db_url`. If the database doesn't exist, it will not automatically create it for you.
 
+```bash
+SET GLOBAL default_storage_engine = 'InnoDB';
+CREATE DATABASE DB_NAME CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
```

Comment:
Why not set the `ENGINE` directly on the create database query? Setting a global seems a bit odd?

Reply:
You can not set the engine on database level only on table level so it could only be done in the internal initialisation progress in the App.  Normally InnoDb should be the default engine any way but it wasnt on my device installing a mariaDB 10.

alternative we could just make it a comment 

"check that your engine is set to InnoDB"

I had a lot of errors during initialisation because it was still myisam ( which is quite strange vor mariaDB 10.x).

Reply:
> You can not set the engine on database level only on table level

Oh right 🤦 

I wonder if this should be documented like this, or ensures in our recorder so it creates tables with the right engine.

The character set and collation actually are set by the recorder on table create, so not needed here.

Reply:
The better way would be to ensure InnoDb during setup I guess, but im quite new to HA and just wanted to help others which might run into the same setup.

maybe we only add an Information to make sure the default is InnoDB for now?

Reply:
I've discussed this with one of the core maintainers the regularly works on the recorder.
We don't set the engine, as some use the aria or extradb engines (which also work). We can't change that at this point without being breaking.

I think, for that reason, this addition is fine.

Thanks @TheBelgarion  👍